YAXLib/SerializerOptions.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (C) Sina Iravanian, Julian Verdurmen, axuno gGmbH and other contributors.
+// Licensed under the MIT license.
+
+using System;
+using System.Xml.Linq;
+
+namespace YAXLib
+{
+    /// <summary>
+    ///     Settings to influence the process of serialization or de-serialization of <see cref="YAXSerializer"/>s.
+    /// </summary>
+    public class SerializerOptions
+    {
+        private const int DefaultMaxRecursion = 300;
+
+        /// <summary>
+        ///     Creates a new instances of settings to influence the process of
+        ///     serialization or de-serialization of <see cref="YAXSerializer"/>s.
+        /// </summary>
+        public SerializerOptions()
+        {
+            MaxRecursion = DefaultMaxRecursion;
+
+            // Initialization with compatibility to v2.x:
+
+            ExceptionBehavior = YAXExceptionTypes.Warning;
+            ExceptionHandlingPolicies = YAXExceptionHandlingPolicies.ThrowErrorsOnly;
+            SerializationOptions = YAXSerializationOptions.DontSerializeNullObjects;
+            AttributeName = new YAXAttributeName {Dimensions = "dims", RealType = "realtype"};
+            Namespace = new YAXNameSpace {Prefix = "yaxlib", Uri = XNamespace.Get("http://www.sinairv.com/yaxlib/")};
+        }
+
+        /// <summary>
+        /// Definitions for special attribute names.
+        /// </summary>
+        public class YAXAttributeName
```

Comment:
I would not recommend this. 

> ❌ DO NOT use public nested types as a logical grouping construct; use namespaces for this.

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/nested-types